libs/utils/src/properties.c

```diff
@@ -191,324 +476,423 @@ static void parseLine(const char* line, celix_properties_t *props) {
+                                                   const char* key,
+                                                   const celix_version_t* defaultValue) {
+    celix_properties_entry_t* entry = celix_properties_getEntry(properties, key);
+    if (entry != NULL && entry->valueType == CELIX_PROPERTIES_VALUE_TYPE_VERSION) {
+        return entry->typed.versionValue;
     }
+    return defaultValue;
 }
 
-double celix_properties_getAsDouble(const celix_properties_t *props, const char *key, double defaultValue) {
-    double result = defaultValue;
-    const char *val = celix_properties_get(props, key, NULL);
-    if (val != NULL) {
-        char *enptr = NULL;
-        errno = 0;
-        double r = strtod(val, &enptr);
-        if (enptr != val && errno == 0) {
-            result = r;
+celix_version_t* celix_properties_getAsVersion(const celix_properties_t* properties,
+                                               const char* key,
+                                               const celix_version_t* defaultValue) {
+    celix_properties_entry_t* entry = celix_properties_getEntry(properties, key);
+    if (entry != NULL && entry->valueType == CELIX_PROPERTIES_VALUE_TYPE_VERSION) {
+        return celix_version_copy(entry->typed.versionValue);
+    }
+    if (entry != NULL && entry->valueType == CELIX_PROPERTIES_VALUE_TYPE_STRING) {
+        celix_version_t* createdVersion = celix_version_createVersionFromString(entry->value);
+        if (createdVersion != NULL) {
+            return createdVersion;
         }
     }
-    return result;
+    return defaultValue == NULL ? NULL : celix_version_copy(defaultValue);
 }
 
-void celix_properties_setDouble(celix_properties_t *props, const char *key, double val) {
-    char buf[32]; //should be enough to store long long int
-    int writen = snprintf(buf, 32, "%f", val);
-    if (writen <= 31) {
-        celix_properties_set(props, key, buf);
-    } else {
-        fprintf(stderr,"buf to small for value '%f'\n", val);
-    }
+celix_status_t
+celix_properties_setVersion(celix_properties_t* properties, const char* key, const celix_version_t* version) {
+    return celix_properties_createAndSetEntry(properties, key, NULL, NULL, NULL, NULL, celix_version_copy(version));
 }
 
-bool celix_properties_getAsBool(const celix_properties_t *props, const char *key, bool defaultValue) {
-    bool result = defaultValue;
-    const char *val = celix_properties_get(props, key, NULL);
-    if (val != NULL) {
-        char buf[32];
-        snprintf(buf, 32, "%s", val);
-        char *trimmed = celix_utils_trimInPlace(buf);
-        if (strncasecmp("true", trimmed, strlen("true")) == 0) {
-            result = true;
-        } else if (strncasecmp("false", trimmed, strlen("false")) == 0) {
-            result = false;
+void celix_properties_setVersionWithoutCopy(celix_properties_t* properties, const char* key, celix_version_t* version) {
```

Comment:
`key` will be copied. 
Will our users confuse it with `celix_properties_setWithoutCopy`? In that case, `key` will NOT be copied.